src/lib/bootstrap.js

```diff
@@ -7,18 +7,18 @@
 
 import config from "./bootstrap-config";
 import "@webcomponents/webcomponentsjs/webcomponents-loader.js";
+import {router, entrypointLoaded} from "./router";
+import entrypointForRoute from "./entrypoint-for-route";
 
 console.info("web.dev", config.version);
 
 WebComponents.waitFor(async () => {
-  return new Promise((resolve, reject) => {
-    // nb. import() is fairly well supported (although not as much as raw modules), but we just
-    // don't need it
-    const s = document.createElement("script");
-    s.type = "module";
-    s.onerror = reject;
-    s.onload = () => resolve();
-    s.src = "/app.js";
-    document.head.appendChild(s);
-  });
+  // ... only load the first entrypoint if the user hasn't changed URLs in the meantime
+  if (!entrypointLoaded) {
```

Comment:
Did you add this here because the router returns if `isFirstRun` is true?
I was thinking that this work should live entirely in the router if possible.

Reply:
I answered this down below but I'll put it again here (oops).

Routing doesn't need Web Components or other site code to be ready—it's self-contained. And the "complexity" is a single conditional statement with a bunch of comments.